valohai_yaml/pipelines/conversion.py

```diff
@@ -60,8 +93,15 @@ def convert_executionlike_node(self, node: Union[ExecutionNode, TaskNode]) -> Co
         if not step:  # pragma: no cover
             raise ValueError(f"Step {step_name} not found in {self.config}")
         step_data = step.serialize()
-        step_data.update(override)
 
+        if node_data.get("mode"):
```

Comment:
Just look at the node directly :)
```suggestion
        if node.override_mode == OverrideMode.MERGE:
```
or similar.